src/client/dcp_client/gui/napari_window.py

```diff
@@ -22,33 +50,208 @@ def __init__(self, app: Application):
+
+        self.active_mask = True
+    
+    def switch_to_non_active_mask(self):
+
+        self.instances = set(np.unique(self.layer.data[self.active_mask_index])[1:])
+
+        self.switch_controls("paint_button", False)
+        self.switch_controls("erase_button", False)
+        self.switch_controls("fill_button", True) 
+
+        self.active_mask = False
+
+    def set_active_mask(self):
+        self.mask_choice_dropdown.setDisabled(True)
+        self.active_mask_index = self.mask_choice_dropdown.currentData()
+        self.instances = set(np.unique(self.layer.data[self.active_mask_index])[1:])
+        self.prev_mask = self.layer.data[self.active_mask]
+        if self.active_mask_index == 1:
+            self.switch_to_non_active_mask()
+
+    def find_edges(self, idx=None):
+        '''
+        idx - indices of the specific labels from which to get contour
+        '''
+        if idx is not None and not isinstance(idx, list):
+            idx = [idx]
+
+        active_mask = self.layer.data[self.active_mask_index]
+
+        instances = np.unique(active_mask)[1:]
+        edges = np.zeros_like(active_mask).astype(int)
+
+        # to merge the discontinuous contours
+        kernel = np.ones((5, 5))
+
+        if len(instances):
+            for i in instances:
+                if idx is None or i in idx:
+        
+                    mask_instance = (active_mask == i).astype(np.uint8)
+
+                    edge_mask = 255 * (canny(255 * (mask_instance)) > 0).astype(np.uint8)
+                    edge_mask = cv2.morphologyEx(
+                        edge_mask, 
+                        cv2.MORPH_CLOSE, 
+                        kernel,
+                    )
+                    edges = edges + edge_mask
+
+            # if masks are intersecting then we want to count it only once
+            edges = edges > 0
+            # cut the contours
+            self.layer.data = self.layer.data * np.invert(edges).astype(np.uint8)
+            
+    def copy_mask_callback(self, layer, event):
+
+        source_mask = layer.data
+    
+        if event.type == "mouse_press":
+
+            c, event_x, event_y = event.position
+            c, event_x, event_y = int(c), int(np.round(event_x)), int(np.round(event_y))
+
+            if source_mask[c, event_x, event_y] == 0:
+                self.new_pixel = True
+            else:
+                self.new_pixel = False
+
+            self.event_coords = (c, event_x, event_y)
+
+           
+
+        elif event.type == "set_data":
+            
+            active_mask_current = self.active_mask
+
+            if self.viewer.dims.current_step[0] == self.active_mask_index:
+                self.switch_to_active_mask()
+            else:
+                self.switch_to_non_active_mask()
+
+            if self.event_coords is not None:
+
+                c, event_x, event_y = self.event_coords
+                
+                if c == self.active_mask_index:
```

Comment:
from here on, I'm having trouble understanding what's going on. Try to either add more comments or make the code more explicit. Let me know if you need more feedback :)